### PR TITLE
Mccalluc/exponential explosion

### PR DESCRIPTION
**Pull request by mccalluc**
There are bugs and a lack of testing on the flat-file imports. This adds some tests... and I'd hope fixing the bugs might be easy, but it hasn't been obvious, yet.

### CODE REVIEW

File: refinery/data_set_manager/tests.py [hunk 1]
```diff
@@ -1958,3 +1960,34 @@ def test_post_bad_isa_tab_file(self):
...
+            'test-data', 'single-file', filename
+        )
+        with open(path, 'r') as f:
+            process_metadata_table(
+                username='guest',
+                title='fake',
+                metadata_file=f,
+                source_columns=[0],
+                data_file_column=2,
+            )
```
Comment (scottx611x) [inline refinery/data_set_manager/tests.py]: Assertions here regarding created objects?

**Comment by mccalluc** [comment]
The fix probably will not be in before we freeze... but it might be good to merge a PR even if it just surfaces a problem in a new test, even if we can't fully resolve it right now?

**Comment by mccalluc** [comment]
@scottx611x : Here's something.
- Just put some duck tape on the false positive exponential nodes problem: Still don't understand what's actually going on.
- Single line CSV ingests now, and we assert that we get the right number of nodes out.
- Added a test of `update_annotated_nodes` where the problem really was... though the behavior there isn't what I'd expect.

File: refinery/data_set_manager/tests.py (hunk 1 above)
Comment (mccalluc) [reply refinery/data_set_manager/tests.py]: see below

**Pull request merged**